// subscene/lib/getSubLink.js
var searchTitle=require('./searchTitle.js');
var searchRelease=require('./searchRelease');
var chalk=require('chalk');
var getZippedSub=require('./zip/sub.js');
var langCodes=require('./langCodes.js');
var scrapDownloadLink=require("./scrapDownloadLink.js");
var $=require('cheerio');
var queryCheck=require('./query.js');

var i=0;
var getSubtitleInterfaceLink = require('./getSubtitleInterfaceLink.js');
function getsubtitleLink(movie,lang,locationSave,mainCallback,cb){

queryCheck(movie,lang,function(err,searchType,response,body,callback){

  if(err){
    mainCallback(err);
    return;
  }

if(searchType==="title"){

  searchTitle(response,body,lang,function(err, link){
  if(err){
    mainCallback(err,"data");
    return;
  }

  getSubtitleInterfaceLink(link,lang,function(err,subLink){
    if(err){
      mainCallback(true);
      return;

    }
    scrapDownloadLink(subLink,function(error,downloadLink){
      if(error){
        mainCallback(true);
        return;
      }


      getZippedSub(movie,downloadLink,locationSave,function(err,fileName,zippedData){
      mainCallback(err,fileName,zippedData);


});



  });

});

});

}

else{
  searchRelease(response,body,lang,function(err,link){
    if(err){

      mainCallback(true);
      return;
    }

    scrapDownloadLink(link,function(errorr,downloadLink){
      if(errorr){
        mainCallback(true);
        return;
      }

      getZippedSub(movie,downloadLink,locationSave,function(err,fileName,zippedData){
      mainCallback(err,fileName,zippedData);

});

    });

  });


}

});
};




module.exports=getsubtitleLink;
